### PR TITLE
chore: support lazy materialization for more queries.

### PR DESCRIPTION
**Pull request by RinChanNOWWW**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

- Rename `lazy_topn_threshold` to `lazy_read_threshold`.
- Support lazy materialization for where-limit clause.

Besides queries with `ORDER BY` and `LIMIT`,  queries with `WHERE` and `LIMIT` will also use lazy materialization now.

### CODE REVIEW

File: docs/doc/13-sql-reference/20-system-tables/system-settings.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ SELECT * FROM system.settings;
...
 | hide_options_in_show_create_table     | 1            | 1            | SESSION | Hides table-relevant information, such as SNAPSHOT_LOCATION and STORAGE_FORMAT, at the end of the result of SHOW TABLE CREATE.                                                      | UInt64 |
 | input_read_buffer_size                | 1048576      | 1048576      | SESSION | Sets the memory size in bytes allocated to the buffer used by the buffered reader to read data from storage.                                                                        | UInt64 |
-| lazy_topn_threshold                   | 1000         | 1000         | SESSION | Enable lazy materialization and set the limit threshold of Top-N queries. Set the value to 0 to disable this setting.                                                               | UInt64 |
+| lazy_read_threshold                   | 1000         | 1000         | SESSION | Enable lazy materialization and set the limit threshold of queries with limit. Set the value to 0 to disable this setting.                                                               | UInt64 |
```
Comment (soyeric128) [inline docs/doc/13-sql-reference/20-system-tables/system-settings.md:25]:
```suggestion
| lazy_read_threshold                   | 1000         | 1000         | SESSION | Sets the maximum LIMIT in a query to enable lazy read optimization. Setting it to 0 disables the optimization.                                                               | UInt64 |
```

File: tests/sqllogictests/suites/mode/standalone/explain/lazy_read.test [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+set lazy_read_threshold=100
+
+statement ok
+drop table if exists t_lazy
+
+statement ok
+create table t_lazy (a int, b float, c string, d tuple(a int, b int), e date)
+
+query T
+explain select * from t_lazy order by a desc limit 2
```
Comment (BohuTANG) [inline tests/sqllogictests/suites/mode/standalone/explain/lazy_read.test:11]: What's the explain when `set lazy_read_threshold=0`?

Comment (RinChanNOWWW) [reply tests/sqllogictests/suites/mode/standalone/explain/lazy_read.test:11]: There will not be `RowFetch` if `lazy_read_threshold` is set to 0.

**Review by sundy-li — APPROVED** [review]

File: tests/sqllogictests/suites/mode/standalone/explain/lazy_read.test (hunk 2 above)
Comment (BohuTANG) [reply tests/sqllogictests/suites/mode/standalone/explain/lazy_read.test:11]: Can we add the case to the test, it's very useful to know what's the difference from the explain.

Comment (RinChanNOWWW) [reply tests/sqllogictests/suites/mode/standalone/explain/lazy_read.test:11]: Ok

**Pull request merged**